react-app/src/redux/game/actions.js

```diff
@@ -24,6 +32,81 @@ export const updateStatus = status => ({
   status
 });
 
+const updateGameActions = {
+  updateGameSuccess: (data, obj, dispatch) => {
+    const id = data.id;
+    const gameCount = Number(data.gameCount) + 1;
+    let objUpdate = { gameCount };
+    if (obj.winner && obj.winner === USER_PLAYER_MARK) {
+      const points = Number(data.points) + obj.points;
+      const historyPoints = { ...data.historyPoints, [new Date().getTime()]: obj.points };
+      objUpdate = { ...objUpdate, points, historyPoints };
+    }
+    authService.setUser(id, objUpdate);
+    dispatch({
+      type: actionNames.UPDATE_GAME_SUCCESS,
+      loading: false,
+      ...objUpdate
+    });
+  },
+  updateGameFailure: (message, dispatch) => {
+    dispatch({
+      type: actionNames.UPDATE_GAME_FAILURE,
+      loading: false,
+      message
+    });
+  }
+};
+
+const getUserPointsActions = {
+  getUserPointsSuccess: (data, dispatch) => {
+    const gameCount = Number(data.gameCount);
+    const points = Number(data.points);
+    const historyPoints = data.historyPoints;
+    dispatch({
+      type: actionNames.GET_USER_POINT_SUCCESS,
+      loading: false,
+      gameCount,
+      points,
+      historyPoints
+    });
+  },
+  getUserPointsFailure: (message, dispatch) => {
+    dispatch({
+      type: actionNames.GET_USER_POINT_FAILURE,
+      loading: false,
+      message
+    });
+  }
+};
+
+export const updateGame = obj => async dispatch => {
+  dispatch({
+    type: actionNames.LOADING
+  });
+  dispatch(toggleMark(obj.history, obj.stepNumber, obj.xIsNext, obj.winner));
+  if (obj.winner || obj.stepNumber === 9) {
+    const response = await authService.getUser({ token: obj.token });
+    if (response && response.ok && response.data.length > 0) {
+      updateGameActions.updateGameSuccess(response.data[0], obj, dispatch);
```

Comment:
Dispatches should be done here instead. Private action creators shouldnt contain a dispatch